qiskit/circuit/random/utils.py

```diff
@@ -118,6 +86,256 @@ def random_circuit(
+    from the set of standard gates in :mod:`qiskit.circuit.library.standard_gates`.
+    User can attach a float value indicating the probability of getting selected as a
+    metadata to the edge of the graph generated. Pass in None if the probability is
+    not available. Even if a single probability is None, this will turn off the probabilistic
+    selection of qubit-pair.
+
+    If :arg:`max_operands` is set to 2, then a 2Q gate is chosen at random and a qubit-pair is
+    also chosen at random based on the probability attached to the qubit-pair, this 2Q gate
+    is applied to that qubit-pair and the rest of the qubits are filled with 1Q operations
+    based on :arg:`insert_1q_oper` is set to True. This makes a single layer of the cirucit.
+
+    Example:
+
+    .. plot::
+       :include-source:
+
+       from qiskit.circuit.random.utils import random_circuit_with_graph
+       import rustworkx as rx
+       pydi_graph = rx.PyDiGraph()
+       pydi_graph.add_nodes_from(range(10))
+       cp_map = [
+       (0, 2, 0.1),
+       (1, 3, 0.15),
+       (2, 4, 0.15),
+       (3, 4, 0.1),
+       (5, 7, 0.13),
+       (4, 7, 0.07),
+       (7, 9, 0.1),
+       (5, 8, 0.1),
+       (6, 9, 0.1),
+       ]
+
+       pydi_graph.add_edges_from(cp_map)
+       inter_graph = (pydi_graph, None, None, None)
+       qc = random_circuit_with_graph(inter_graph, depth=3)
+       qc.draw(output='mpl')
+
+    Args:
+        interaction_graph (int): Interaction Graph
+        depth (int): Minimum number of times every qubit-pair is must be used .
+        max_operands (int): maximum qubit operands of each gate (between 1 and 2)
+        measure (bool): if True, measure all qubits at the end
+        conditional (bool): if True, insert middle measurements and conditionals
+        reset (bool): if True, insert middle resets
+        seed (int): sets random seed (optional)
+        insert_1q_oper (bool): Insert 1Q operations to the qubits which are left after
+        applying a 2Q gate on the selected qubit-pair.
+        prob_conditional (float): Probability less than 1.0, this is used to control the
+        occurence of conditionals in the circuit.
+
+    Returns:
+        QuantumCircuit: constructed circuit
+
+    Raises:
+        CircuitError: when invalid options given
+        ValueError: when the probabilities for qubit-pairs are passed and they do not add to 1.0
+        and when a negative probability exists.
+    """
+    pydi_graph, pydi_graph_node_map, _, _ = interaction_graph
+    edges_probs = pydi_graph.edges()
+
+    # Just a switch for the probability weighted selection of qubit-pairs.
+    # If any value of the probability is None or the whole probability list is None,
+    # then, the probability weighted selection of qubit-pair would be turned off.
+    prob_weighted_mapping = not None in edges_probs
+
+    if prob_weighted_mapping:
+        for prob in edges_probs:
```

Comment:
edge weights in interaction_graph must be positive and should be normalized to sum up to 1.0